src/file/format/ini.rs

```diff
@@ -3,35 +3,45 @@ use std::error::Error;
 use ini::Ini;
 
 use crate::map::Map;
-use crate::value::{Value, ValueKind};
+use crate::value::{Table, Value, ValueKind};
 
 pub fn parse(
     uri: Option<&String>,
     text: &str,
 ) -> Result<Map<String, Value>, Box<dyn Error + Send + Sync>> {
-    let mut map: Map<String, Value> = Map::new();
-    let i = Ini::load_from_str(text)?;
-    for (sec, prop) in i.iter() {
-        match sec {
-            Some(sec) => {
-                let mut sec_map: Map<String, Value> = Map::new();
-                for (k, v) in prop.iter() {
-                    sec_map.insert(
-                        k.to_owned(),
-                        Value::new(uri, ValueKind::String(v.to_owned())),
-                    );
-                }
-                map.insert(sec.to_owned(), Value::new(uri, ValueKind::Table(sec_map)));
-            }
-            None => {
-                for (k, v) in prop.iter() {
-                    map.insert(
-                        k.to_owned(),
-                        Value::new(uri, ValueKind::String(v.to_owned())),
-                    );
-                }
-            }
-        }
+    let value = from_ini(uri, Ini::load_from_str(text)?);
+
+    match value.kind {
+        ValueKind::Table(map) => Ok(map),
+
+        _ => Ok(Map::new()),
+    }
```

Comment:
How about something like

```rust
if let ValueKind::Table(map) = value.kind {
    Ok(map)
} else {
    Ok(Map::new)
}
```

here?

Reply:
I used what you have for every other format atm. A separate PR I referenced provides a common method for this logic based on a TODO you have in each of the files.

Thus if you approve of that approach, it'd be better to just rebase here and switch to the method?

Reply:
Agreed